test/acceptance/list_test.exs

```diff
@@ -132,6 +133,94 @@ defmodule Acceptance.ListTest do
       assert as_html(markdown) == {:ok, html, messages}
     end
 
+    test "tight lists have no paragraph wrappers" do
+      markdown = """
+      - a
+      - b
+      """
+      html = """
+      <ul>
+      <li>a
+      </li>
+      <li>b
+      </li>
+      </ul>
+      """
+      messages = []
+
+      assert as_html(markdown) == {:ok, html, messages}
+    end
+
+    test "loose lists have paragraph wrappers" do
+      # Here the blank line makes this a loose list
+      markdown = """
+      - a
+
+      - b
+      """
+      html = """
+      <ul>
+      <li><p>a</p>
+      </li>
+      <li><p>b</p>
+      </li>
+      </ul>
+      """
+      messages = []
+
+      assert as_html(markdown) == {:ok, html, messages}
+    end
+
+    test "lists with nested content are still tight without blank lines" do
+      # Here the outer and inner lists are both tight
+      markdown = """
+      - a
+        - b
+      """
+      html = """
+      <ul>
+      <li>a
+      <ul>
+      <li>b
+      </li>
+      </ul>
+      </li>
+      </ul>
+      """
+      messages = []
+
+      assert as_html(markdown) == {:ok, html, messages}
+    end
+
+    test "sublist can be loose or tight irrespective of parent" do
+      # Here the outer list is tight and the inner list is loose
+      markdown = """
+      - a
+        - b
+
+          c
+      - d
+      """
+      # TODO: There should be no leading whitespace before 'c'. But it seems
+      # unrelated to loose/tight lists.
+      html = """
+      <ul>
+      <li>a
+      <ul>
+      <li><p>b</p>
+      <p>  c</p>
```

Comment:
This was a bit unexpected.  Seems like an issue with the bullet being stripped for 'b', but nothing being stripped on 'c'.  I'm not positive, but I think it's unrelated to tight/loose lists.

Reply:
absolutely e.g.

```elixir
# from master
Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> x="""
...(1)> - a
...(1)>
...(1)>   b
...(1)> """
"- a\n\n  b\n"
iex(2)> Earmark.as_html! x
"<ul>\n<li><p>a</p>\n<p>  b</p>\n</li>\n</ul>\n"
```
The whitespace for sure is not _pretty_ but invisible, so maybe a minor bug report on this?

Reply:
Hmm or maybe it is even _correct_ ?

```elixir
iex(5)> x="""
...(5)> - a
...(5)> b
...(5)> """
"- a\nb\n"
iex(6)> Earmark.as_html! x
"<ul>\n<li>a\nb\n</li>\n</ul>\n"
```
Never mind, absolutely unrelated

Reply:
Okay, I think a minor bug, too.  You're right, it seems totally invisible in the browser.  I was thinking the browser would collapse the whitespace and keep one of the two spaces, but it doesn't seem to keep any, so that's good.